concordium-consensus/src/Concordium/GlobalState/Persistent/CachedRef.hs

```diff
@@ -344,6 +344,32 @@ data HashedCachedRef' h c a
         hcrHash :: !h
     }
 
+-- |Migrate a 'HashedCachedRef'. This
+migrateHashedCachedRef' ::
```

Comment:
Maybe we could have a class  `SupportsCache c a...` =  (MonadCache c m,
      BlobStorable m a,
      Cache c,
      CacheKey c ~ BlobRef a,
      CacheValue c ~ a)

Reply:
I do not think it is worth it for this PR. While the constraint here looks big, it is easy to understand I think.

Such a supportsCache constraint should be used elsewhere, but I don't think I want to add to this PR those changes.